assembly/assembly-main/src/assembly/bin/launcher.sh

```diff
@@ -0,0 +1,126 @@
+#!/bin/sh
+
+# See: https://sipb.mit.edu/doc/safe-shell/
+set -e
+#set +o posix
+
+# Run the finish function if exit signal initiated
+trap exit SIGHUP SIGINT SIGTERM
```

Comment:
I think you can move `set -e` and `trap` commands down to other commands. It will make reading easier